cli/common/tarantool.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	tarantoolVersionRegexp *regexp.Regexp
+)
+
+func init() {
+	tarantoolVersionRegexp = regexp.MustCompile(`\d+\.\d+\.\d+-\d+-\w+`)
+}
+
+// GetTarantoolDir returns Tarantool executable directory
+func GetTarantoolDir() (string, error) {
+	var err error
+
+	tarantool, err := exec.LookPath("tarantool")
+	if err != nil {
+		return "", fmt.Errorf("tarantool executable not found")
+	}
+
+	return filepath.Dir(tarantool), nil
+}
+
+// TarantoolIsEnterprise checks if Tarantool is Enterprise
+func TarantoolIsEnterprise(tarantoolDir string) (bool, error) {
+	var err error
+
+	tarantool := filepath.Join(tarantoolDir, "tarantool")
+	versionCmd := exec.Command(tarantool, "--version")
+
+	tarantoolVersion, err := GetOutput(versionCmd, nil)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.HasPrefix(tarantoolVersion, "Tarantool Enterprise"), nil
+}
+
+// GetTarantoolVersion gets Tarantool version
+func GetTarantoolVersion(tarantoolDir string) (string, error) {
+	var err error
+
+	tarantool := filepath.Join(tarantoolDir, "tarantool")
+	versionCmd := exec.Command(tarantool, "--version")
+
+	tarantoolVersion, err := GetOutput(versionCmd, nil)
+	if err != nil {
+		return "", err
+	}
+
+	tarantoolVersion = tarantoolVersionRegexp.FindString(tarantoolVersion)
+
+	if tarantoolVersion == "" {
+		return "", fmt.Errorf("Failed to match Tarantool version")
+	}
+
+	return tarantoolVersion, nil
+}
+
+// GetNextMajorVersion computes next major version for a given one.
+// For example, for 1.10.3 it's 2
+func GetNextMajorVersion(version string) (string, error) {
```

Comment:
May be you can use "Segments" from go-version - https://github.com/hashicorp/go-version/blob/master/version.go#L330

Reply:
Yes, thx.

Done.